src/test/java/com/uid2/admin/managers/KeysetManagerTest.java

```diff
@@ -0,0 +1,133 @@
+package com.uid2.admin.managers;
+
+import com.uid2.admin.secret.IKeysetKeyManager;
+import com.uid2.admin.store.writer.KeysetStoreWriter;
+import com.uid2.admin.util.KeysetUtil;
+import com.uid2.shared.auth.ClientKey;
+import com.uid2.shared.auth.Keyset;
+import com.uid2.shared.auth.KeysetSnapshot;
+import com.uid2.shared.auth.Role;
+import com.uid2.shared.store.reader.RotatingKeysetProvider;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class KeysetManagerTest {
+    protected AutoCloseable mocks;
+
+    @Mock protected KeysetStoreWriter keysetStoreWriter;
+    @Mock protected RotatingKeysetProvider keysetProvider;
+    @Mock
+    protected IKeysetKeyManager keysetKeyManager;
+    @Mock protected KeysetSnapshot keysetSnapshot;
+
+    @BeforeEach
+    public void setupMocks() {
+        mocks = MockitoAnnotations.openMocks(this);
+        when(keysetProvider.getSnapshot()).thenReturn(keysetSnapshot);
+    }
+
+    @AfterEach
+    public void teardown() throws Exception {
+        mocks.close();
+    }
+
+    protected void setKeysets(Map<Integer, Keyset> keysets) {
+        when(keysetSnapshot.getAllKeysets()).thenReturn(keysets);
```

Comment:
Can we please use a real class for keysetSnapshot instead of mocking it? It is easy to instantiate a real keysetSnapshot (I made this change in Operator). getAllKeysets() already returns the keysets you passed into the ctor.